internal/tests/integration/commands/hset_test.go

```diff
@@ -0,0 +1,75 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHSet(t *testing.T) {
+	exec, err := NewHTTPCommandExecutor()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testCases := []TestCase{
+		{
+			Name: "HSET with simple key value pairs in the hash",
+			Commands: []HTTPCommand{
+				{Command: "HSET", Body: []string{"user", "name", "John Doe", "age", "30"}},
+			},
+			Result: []TestCaseResult{
+				{Expected: "2"},
+			},
+		},
+		{
+			Name: "HSET update one key and set a new key",
+			Commands: []HTTPCommand{
+				{Command: "HSET", Body: []string{"user1", "name", "John Doe", "age", "30"}},
+				{Command: "HSET", Body: []string{"user1", "name", "John Loe", "gender", "Male"}},
+			},
+			Result: []TestCaseResult{
+				{Expected: "2"},
+				{Expected: "1"},
+			},
+		},
+		{
+			Name: "HSET with invalid number of arguments",
+			Commands: []HTTPCommand{
+				{Command: "HSET", Body: []string{"user", "name", "John Loe", "gender"}},
+			},
+			Result: []TestCaseResult{
+				{ErrorExpected: true},
```

Comment:
Can we assert on the error string which we receive?
This would help us maintain sanity for API responses for errors.

Reply:
Sure, will do!